crates/swc_ecma_transforms_react/src/refresh/mod.rs

```diff
@@ -246,12 +246,37 @@ where
 
         let mut should_refresh = self.should_reset;
         if let Some(comments) = &self.comments {
-            if n.hi != BytePos(0) {
-                comments.with_leading(n.hi - BytePos(1), |comments| {
-                    if comments.iter().any(|c| c.text.contains("@refresh reset")) {
-                        should_refresh = true
-                    }
-                });
+            comments.with_leading(n.hi - BytePos(1), |comments| {
+                if comments.iter().any(|c| c.text.contains("@refresh reset")) {
+                    should_refresh = true
+                }
+            });
+
+            if should_refresh {
```

Comment:
I tried to early escape if any of comment have directives. However not much satisfied around code itself if repeated logics can be deduped or not.

Reply:
I think it's fine for now.